src/libcore/slice.rs

```diff
@@ -1212,6 +1212,9 @@ impl<'a, T> Iterator for Windows<'a, T> {
     }
 }
 
+#[stable]
+impl<'a, T> ExactSizeIterator for Windows<'a, T> {}
```

Comment:
For now, yes, let's remove it until we're guaranteed that `size_hint` always returns an exact size hint.